src/bootstrap/bootstrap.py

```diff
@@ -546,51 +557,30 @@ def build_triple(self):
-            self.get_mk('CFG_DISABLE_MANAGE_SUBMODULES') == "1":
+                self.get_toml('submodules') == "false" or \
+                self.get_mk('CFG_DISABLE_MANAGE_SUBMODULES') == "1":
             return
 
         print('Updating submodules')
-        output = self.output(["git", "submodule", "status"], cwd=self.rust_root)
-        submodules = []
-        for line in output.splitlines():
-            # NOTE `git submodule status` output looks like this:
-            #
-            # -5066b7dcab7e700844b0e2ba71b8af9dc627a59b src/liblibc
-            # +b37ef24aa82d2be3a3cc0fe89bf82292f4ca181c src/compiler-rt (remotes/origin/..)
-            #  e058ca661692a8d01f8cf9d35939dfe3105ce968 src/jemalloc (3.6.0-533-ge058ca6)
-            #
-            # The first character can be '-', '+' or ' ' and denotes the
-            # `State` of the submodule Right next to this character is the
-            # SHA-1 of the submodule HEAD And after that comes the path to the
-            # submodule
-            path = line[1:].split(' ')[1]
-            submodules.append([path, line[0]])
-
-        self.run(["git", "submodule", "sync"], cwd=self.rust_root)
-
-        for submod in submodules:
-            path, status = submod
-            if path.endswith('llvm') and \
-                (self.get_toml('llvm-config') or self.get_mk('CFG_LLVM_ROOT')):
-                continue
-            if path.endswith('jemalloc') and \
-                (self.get_toml('jemalloc') or self.get_mk('CFG_JEMALLOC_ROOT')):
-                continue
-            submod_path = os.path.join(self.rust_root, path)
-
-            if status == ' ':
-                self.run(["git", "reset", "--hard"], cwd=submod_path)
-                self.run(["git", "clean", "-fdx"], cwd=submod_path)
-            elif status == '+':
-                self.run(["git", "submodule", "update", path], cwd=self.rust_root)
-                self.run(["git", "reset", "--hard"], cwd=submod_path)
-                self.run(["git", "clean", "-fdx"], cwd=submod_path)
-            elif status == '-':
-                self.run(["git", "submodule", "init", path], cwd=self.rust_root)
-                self.run(["git", "submodule", "update", path], cwd=self.rust_root)
-            else:
-                raise ValueError('unknown submodule status: ' + status)
+        self.run(["git", "submodule", "-q", "sync"], cwd=self.rust_root)
+        # FIXME: nobody does, but this won't work well with whitespace in
+        # submodule path
+        submodules = [s.split()[1] for s in subprocess.check_output(
+            ["git", "config", "--file", os.path.join(
+                self.rust_root, ".gitmodules"), "--get-regexp", "path"]).splitlines()]
+        submodules = [module for module in submodules
+                      if not ((module.endswith(b"llvm") and
+                              (self.get_toml('llvm-config') or self.get_mk('CFG_LLVM_ROOT'))) or
+                      (module.endswith(b"jemalloc") and
```

Comment:
It seems that the original PR decoded the pipe output which is in system default encoding, and can vary on Windows. However, I'm quite sure that all code pages are ASCII compat, which doesn't pose a problem if you're dealing with pure English paths. (want some emoji there? :P)